docs/source/tutorial/effect.rst

```diff
@@ -62,3 +61,10 @@ phase, depth, feedback, and delay.
 
 For the constraints of these parameters, please refer to the documentation.
 
+Destroy the effects
+-------------------
+
+Like other objects in palace, effects must be destroyed after it has its jobs
+done with `__exit__()`.  If you use `with` syntax (you should) as mentioned in
```

Comment:
I don't rmb if `__exit__` is instructed to be called explicitly elsewhere in the docs, but let's not do it and keep the thing magical.  Instead, go for `BaseEffect.destroy`.

Reply:
I thought `__exit__()` was just the method's that's called at the end of `with`, like a Python standard?

Reply:
K, I realized that I didn't mention about object destruction before. Hmm how about creating a separate section for `with`? The content should include things about `__enter__()` and `__exit__()`. Then, I'll edit the tutorial for effect to use `with` instead of assigning directly to a variable.

Reply:
> I thought `__exit__()` was just the method's that's called at the end of `with`, like a Python standard?

It is called as the flow get out of the `with` block, but instructing to call it explicitly is a bit weird since it's a dunder/magic method, especially when we have the `destroy` method that does the exact same thing (and is not hinted as magical).  Generally the interface user would only need to call magic methods when they are doing obscure hacks, not in in common use case.

> Hmm how about creating a separate section for `with`?

A dedicated page for context manager usage sounds like a good idea, since pretty much every class we define in palace can be use as a context manager.

Reply:
Ok I'll do it tomorrow because my laptop is running out of battery and the I forgot my converter at school.

Reply:
You must be **@[supakeen](https://github.com/supakeen)** to get to school tomorrow (-;